lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -265,6 +276,63 @@ public void run() {
 		}
 	}
 
+	/**
+	 * Authentication token changes for the current connection are possible when
+	 * the client is in the {@link ConnectionState#connected}, {@link ConnectionState#connecting}
+	 * or {@link ConnectionState#disconnected } states.
+	 *
+	 * In the current protocol version we are not able to update auth params on the fly;
+	 * so disconnect, and the new auth params will be used for subsequent reconnection
+	 * * <p>
+	 * Spec: RTC8 (reauthorise)
+	 * </p>
+	 *
+	 */
+	public void onAuthUpdated() {
+		switch (state.state){
+			case connected:
+			case connecting:
+			case disconnected:
+				connection.once(ConnectionState.failed, new ConnectionStateListener() {
```

Comment:
I don't see why you're depending on events from the `Connection` object here. This code is the `ConnectionManager`, which is responsible to determining the connection state, and issuing events; the `Connection` class is just the client API that exposes those events and state to the application. This class knows what the state is and should be reacting events and changes in its own state, rather than listening for events on the `Connection`.

Also, I have no idea why you're handling this event and then doing a `notify()` on the `ConnectionManager`. Can you explain why?